lib/kafka/cluster.rb

```diff
@@ -46,6 +46,10 @@ def add_target_topics(topics)
         @target_topics.merge(new_topics)
 
         refresh_metadata!
+
+        cluster_info.topics.each do |topic|
+          Protocol.handle_error(topic.topic_error_code)
+        end
```

Comment:
The problem is that "no leader available" is _not_ necessarily an error – it could just be a transient issue. Logging is really the only proper way to handle that situation, aside from using e.g. a timeout to crap out after some time, which I'd prefer not to do.